hpx/runtime/serialization/detail/serialize_collection.hpp

```diff
@@ -0,0 +1,122 @@
+//  Copyright (c) 2017 Anton Bikineev
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef HPX_SERIALIZATION_DETAIL_SERIALIZE_COLLECTION_HPP
+#define HPX_SERIALIZATION_DETAIL_SERIALIZE_COLLECTION_HPP
+
+#include <hpx/config.hpp>
+
+#include <type_traits>
+
+namespace hpx { namespace serialization { namespace detail {
+
+    // default fallbacks. in fact, they should never be called
+    template <class Archive, class T>
+    void save_construct_data(Archive&, T*, unsigned)
+    {
+    }
```

Comment:
Please add a `HPX_ASSERT(false);` if this function is supposed to never be called.